crates/supervisor/storage/src/providers/derivation_provider.rs

```diff
@@ -190,30 +206,36 @@ where
     ) -> Result<(), StorageError> {
         // todo: use cursor to get the last block(performance improvement)
         let latest_block_pair = match self.latest_derived_block_pair() {
-            Ok(pair) => Some(pair),
-            Err(StorageError::EntryNotFound(_)) => None,
+            Ok(pair) => pair,
+            Err(StorageError::EntryNotFound(_)) => return Err(StorageError::DatabaseNotInitialised),
             Err(e) => return Err(e),
         };
 
-        if latest_block_pair.is_some() {
-            let latest_block_pair = latest_block_pair.unwrap();
-            // Validate if the latest derived block is parent of the incoming derived block
-            if !latest_block_pair.derived.is_parent_of(&incoming_pair.derived) {
-                warn!(
-                  target: "supervisor_storage",
-                  latest_derived_block_pair = %latest_block_pair,
-                  incoming_derived_block_pair = %incoming_pair,
-                  "Latest stored derived block is not parent of the incoming derived block"
-                );
-                return Err(StorageError::ConflictError(
-                    "latest stored derived block is not parent of the incoming derived block"
-                        .to_string(),
-                ));
-            }
+        if !latest_block_pair.derived.is_parent_of(&incoming_pair.derived) {
+            warn!(
+              target: "supervisor_storage",
+              latest_derived_block_pair = %latest_block_pair,
+              incoming_derived_block_pair = %incoming_pair,
+              "Latest stored derived block is not parent of the incoming derived block"
+            );
+            return Err(StorageError::ConflictError(
+                "latest stored derived block is not parent of the incoming derived block"
+                    .to_string(),
+            ));
```

Comment:
please add a new error variant for this. keep in mind the supervisor rpc error codes from the spec, this error matches one of those errors exactly, so we probably want to name the variant the name for this error in specs

Reply:
I have added a new variant for this.

Reply:
type safety is better with a variant instead of string messages